### PR TITLE
feat: add dafault value roundrobin for upstream_schema

### PR DESCRIPTION
**Pull request by Hazel6869**
### Description

Fixes  https://github.com/apache/apisix/issues/8114

the docs pr https://github.com/apache/apisix/pull/8191

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by tzssangglass** [comment]
we need to update test cases and docs

**Comment by Hazel6869** [comment]
> we need to update test cases and docs

ok i will do it

File: t/admin/upstream.t [hunk 1]
```diff
@@ -727,3 +727,31 @@ GET /t
...
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "id": 1,
+                    "nodes": {
+                        "127.0.0.1:8080": 1
+                    }
+                }]]
```
Comment (spacewander) [inline t/admin/upstream.t:745]: Better to check the default value of `type` like
https://github.com/apache/apisix/blob/48b84311067ae1b863cfcda23d079a262ff4a8ee/t/admin/upstream.t#L46

**Comment by tzssangglass** [comment]
Do we need to remove a lot of `type = "roundrobin"` in test cases? what's your option? cc @spacewander

File: t/admin/upstream.t [hunk 2]
```diff
@@ -727,3 +727,48 @@ GET /t
...
+                    },
+                    "key": "/apisix/upstreams/admin_up"
+                }]]
+            )
+
+            ngx.status = code
+            ngx.say(body)
+
+            local res = assert(etcd.get('/upstreams/admin_up'))
+            local create_time = res.body.node.value.create_time
```
Comment (spacewander) [inline t/admin/upstream.t:763]: We don't need to check xxx_time, right?

**Comment by spacewander** [comment]
> Do we need to remove a lot of `type = "roundrobin"` in test cases? what's your option? cc @spacewander

Yes, let's do it with script.

File: t/admin/upstream.t (hunk 2 above)
Comment (Hazel6869) [reply t/admin/upstream.t]: Right, I have fixed it.

**Review by tzssangglass — APPROVED** [review]

File: apisix/schema_def.lua [hunk 3]
```diff
@@ -437,6 +437,7 @@ local upstream_schema = {
         type = {
             description = "algorithms of load balancing",
             type = "string",
+            default = "roundrobin",
```
Comment (spacewander) [inline apisix/schema_def.lua:440]: Do we need to update the doc too?

Comment (tokers) [reply apisix/schema_def.lua:440]: +1. cc @Hazel6869

Comment (Hazel6869) [reply apisix/schema_def.lua:440]: I have done. pls see description. @spacewander @tokers

**Review by tokers — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

File: apisix/schema_def.lua (hunk 3 above)
Comment (spacewander) [reply apisix/schema_def.lua:440]: Could you please put the doc in the same PR next time?
So people don't need to check another one.

**Review by spacewander — APPROVED** [review]

File: apisix/schema_def.lua (hunk 3 above)
Comment (Hazel6869) [reply apisix/schema_def.lua:440]: > 

ok

**Pull request merged**